### ISSUE TITLE
Support for external library crates

### ISSUE THREAD

**Issue by korewaChino** [rpm-software-management/mock#976]
Is it possible to support library crates too with mock's external dependencies feature?
https://rpm-software-management.github.io/mock/Feature-external-deps

Currently, external crates are installed using `cargo install` which does not work for library crates. Trying to install a library crate will result in an error due to Cargo not knowing what to do with them.
```
error: there is nothing to install in `zellij-utils v0.31.4`, because it has no binaries
`cargo install` is only for installing programs, and can't be used with libraries.
To use a library crate, add it as a dependency in a Cargo project instead.
```

**Comment by xsuchy** [comment]
Yes, it is possible. But I, personally, do not have time for this. So the best way to get this moving forward is to submit a pull request. 
The relevant code is here https://github.com/rpm-software-management/mock/blob/main/mock/py/mockbuild/external.py#L66
